src/thermo/PengRobinson.cpp

```diff
@@ -0,0 +1,817 @@
+        double pp;
+        return dpdVCalc(T, v, pp);
+    };
+
+    boost::uintmax_t maxiter = 100;
+    std::pair<double, double> vv = bmt::toms748_solve(
+        resid, Vroot[1], Vroot[2], bmt::eps_tolerance<double>(48), maxiter);
+
+    double mmw = meanMolecularWeight();
+    return mmw / (0.5 * (vv.first + vv.second));
+}
+
+double PengRobinson::dpdVCalc(double T, double molarVol, double& presCalc) const
+{
+    double den = molarVol * molarVol + 2 * molarVol * m_b - m_b * m_b;
+    presCalc = GasConstant * T / (molarVol - m_b) - m_aAlpha_mix / den;
+
+    double vpb = molarVol + m_b;
+    double vmb = molarVol - m_b;
+    double dpdv = -GasConstant * T / (vmb * vmb) + 2 * m_aAlpha_mix * vpb / (den*den);
+    return dpdv;
+}
+
+void PengRobinson::calculatePressureDerivatives() const
+{
+    double T = temperature();
+    double mv = molarVolume();
+    double pres;
+
+    m_dpdV = dpdVCalc(T, mv, pres);
+    double vmb = mv - m_b;
+    double den = mv * mv + 2 * mv * m_b - m_b * m_b;
+    m_dpdT = (GasConstant / vmb - daAlpha_dT() / den);
+}
+
+void PengRobinson::updateMixingExpressions()
+{
+    double temp = temperature();
+
+    // Update individual alpha
+    for (size_t j = 0; j < m_kk; j++) {
+        double critTemp_j = speciesCritTemperature(m_a_coeffs(j,j), m_b_coeffs[j]);
+        double sqt_alpha = 1 + m_kappa[j] * (1 - sqrt(temp / critTemp_j));
+        m_alpha[j] = sqt_alpha*sqt_alpha;
+    }
+
+    //Update aAlpha_i, j
+    for (size_t i = 0; i < m_kk; i++) {
+        for (size_t j = 0; j < m_kk; j++) {
+            m_aAlpha_binary(i, j) = sqrt(m_alpha[i] * m_alpha[j]) * m_a_coeffs(i,j);
+        }
+    }
+    calculateAB(m_a,m_b,m_aAlpha_mix);
+}
+
+void PengRobinson::calculateAB(double& aCalc, double& bCalc, double& aAlphaCalc) const
+{
+    bCalc = 0.0;
+    aCalc = 0.0;
+    aAlphaCalc = 0.0;
+    for (size_t i = 0; i < m_kk; i++) {
+        bCalc += moleFractions_[i] * m_b_coeffs[i];
+        for (size_t j = 0; j < m_kk; j++) {
+            double a_vec_Curr = m_a_coeffs(i, j);
+            aCalc += a_vec_Curr * moleFractions_[i] * moleFractions_[j];
+            aAlphaCalc += m_aAlpha_binary(i, j) * moleFractions_[i] * moleFractions_[j];
+        }
+    }
+}
+
+double PengRobinson::daAlpha_dT() const
+{
+    double daAlphadT = 0.0, temp, k, Tc, sqtTr, coeff1, coeff2;
+    for (size_t i = 0; i < m_kk; i++) {
+        // Calculate first derivative of alpha for individual species
+        Tc = speciesCritTemperature(m_a_coeffs(i,i), m_b_coeffs[i]);
+        sqtTr = sqrt(temperature() / Tc); //we need species critical temperature
+        coeff1 = 1 / (Tc*sqtTr);
+        coeff2 = sqtTr - 1;
+        k = m_kappa[i];
+        m_dalphadT[i] = coeff1 * (k*k*coeff2 - k);
+    }
+    //Calculate mixture derivative
+    for (size_t i = 0; i < m_kk; i++) {
+        for (size_t j = 0; j < m_kk; j++) {
+            temp = 0.5 * sqrt((m_a_coeffs(i, i) * m_a_coeffs(j, j)) / (m_alpha[i] * m_alpha[j]));
```

Comment:
Not sure this will impact the failing tests, but you are assuming the user has not provided an `a_ij`, and automatically reverting to the default mixing rules.  This should be:

```suggestion
            temp = 0.5 * m_a_coeffs(i,j) * sqrt(1 / (m_alpha[i] * m_alpha[j]));
```